MusicNotationKit/MusicNotationKit/Measure.swift

```diff
@@ -65,7 +65,7 @@ public struct Measure: ImmutableMeasure, Equatable {
             assertionFailure("note collection should be tuplet, but cast failed")
             throw MeasureError.internalError
         }
-        try tuplet.replaceNote(at: tupletIndex, with: note)
+        try tuplet.replaceNote(at: tuplet.flatIndexes[tupletIndex], with: note)
```

Comment:
I am going to guess this is done this way due to the dependency in modifyTie? Leaving this comment so that we can come back here and update the code once we implement the internal version of `replaceNote(at:)`  at the Measure level.